test/test-poll-oob.c

```diff
@@ -0,0 +1,164 @@
+/* Copyright Fedor Indutny & CurlyMo All rights reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to
+ * deal in the Software without restriction, including without limitation the
+ * rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+ * sell copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+ * IN THE SOFTWARE.
+ */
+
+#if !defined(_WIN32)
+
+#include "uv.h"
+#include "task.h"
+
+#include <errno.h>
+#include <sys/socket.h>
+#include <sys/ioctl.h>
+#include <unistd.h>
+
+static uv_tcp_t server_handle;
+static uv_tcp_t client_handle;
+static uv_tcp_t peer_handle;
+static uv_poll_t poll_req;
+static uv_idle_t idle;
+static uv_os_fd_t client_fd;
+static int ticks;
+static const int kMaxTicks = 10;
+static int check = 0;
+
+static void idle_cb(uv_idle_t* idle) {
+  usleep(100);
+  if (++ticks < kMaxTicks)
+    return;
+
+  uv_poll_stop(&poll_req);
+  uv_close((uv_handle_t*) &server_handle, NULL);
+  uv_close((uv_handle_t*) &client_handle, NULL);
+  uv_close((uv_handle_t*) &peer_handle, NULL);
+  uv_close((uv_handle_t*) idle, NULL);
+}
+
+static void poll_cb(uv_poll_t* handle, int status, int events) {
+  char buffer[5];
+  int n;
+
+  ASSERT(events & UV_PRIORITIZED);
+  n = recv(client_fd, &buffer, 5, MSG_OOB);
+  if(errno == EINVAL) {
+    return;
+  }
+  check = 1;
+  ASSERT(n > 0);
+}
+
+static int non_blocking(int fd, int set) {
+  int r;
+#ifdef FIONBIO
```

Comment:
I'd drop the FIONBIO path; speed is not that important in a test, legibility and maintainability is.